File: fixtures/scale/r/small/textprint.r
#!/usr/bin/env Rscript
# textprint - render styled text blocks in the terminal.

source("config.r")
source("version.r")
source("commands/banner.r")
source("commands/list.r")
source("commands/table.r")

usage <- function() {
  cat("usage: textprint <banner|list|table> [args...]\n")
  cat(sprintf("  textprint %s\n", textprint_version()))
}

main <- function(argv = commandArgs(trailingOnly = TRUE)) {
  command <- if (length(argv) > 0) argv[1] else "help"
  rest <- if (length(argv) > 1) argv[-1] else character(0)
  switch(command,
    banner = cmd_banner(paste(rest, collapse = " ")),
    list = cmd_list(rest),
    table = cmd_table(rest),
    usage()
  )
}

main()

File: fixtures/scale/perl/small/textprint.pl
#!/usr/bin/env perl
use strict;
use warnings;
use lib 'lib';
use TextPrint::ArgParser;
use TextPrint::Renderer;
use TextPrint::InputReader;

my $options = TextPrint::ArgParser::parse(@ARGV);
my $text = TextPrint::InputReader::read_all() || 'a small text printer demo for testing wrap and borders';
my $renderer = TextPrint::Renderer->new($options);
print $renderer->render($text), "\n";
